online-shop/src/app/components/presentational/products-details-view/products-details-view.component.ts

```diff
@@ -0,0 +1,11 @@
+import { Component, Input } from '@angular/core';
+import { ProductDetail } from 'src/app/modules/shared/types/products.types';
+
+@Component({
+  selector: 'app-products-details-view',
+  templateUrl: './products-details-view.component.html',
+  styleUrls: ['./products-details-view.component.scss'],
+})
+export class ProductsDetailsViewComponent {
+  @Input() productDetail: ProductDetail | undefined;
```

Comment:
Can you maybe also add in the optional `?` marker here?

Reply:
done, didn't know I need it.